doc/develop/west/sign.rst

```diff
@@ -7,6 +7,16 @@ The ``west sign`` :ref:`extension <west-extensions>` command can be used to
 sign a Zephyr application binary for consumption by a bootloader using an
 external tool. Run ``west sign -h`` for command line help.
 
+By default ``west sign`` will sign the image assuming that it is running in
+the partition with label ``image-0`` and limit the size of the binary to the
+size of the partition with label ``image-1``.
```

Comment:
Not sure I follow the reasoning for this, the size for image-0 and image-1 should match, why use the size from image-1? If the intention is to use `image size = image-1 start address - image-0 start address` then this is not valid as there is no requirement that image-1 directly follows image-0.

Also, how does this work for applications that have multiple images, e.g. boards with TF-M?

Reply:
The reasoning is in the comments of the python script:
https://github.com/zephyrproject-rtos/zephyr/blob/c9e07fc38515351aebd25d87f3e87f0d3e1dcfc9/scripts/west_commands/sign.py#L341-L346

The point of this change is to support applications with multiple images. Currently if you have a small `image-0` and `image-1` (say a secure application that boots the NS application) and a large `image-2` `image-3` (The actual application), `west sign` will try and sign the large application with the size of the small application, which obviously fails.